dask_ml/wrappers.py

```diff
@@ -662,9 +663,28 @@ def _get_output_dask_ar_meta_for_estimator(model_fn, estimator, input_dask_ar):
     """
     # sklearn fails if input array has size size
     # It requires at least 1 sample to run successfully
-    ar = np.zeros(
-        shape=(1, input_dask_ar.shape[1]),
-        dtype=input_dask_ar.dtype,
-        like=input_dask_ar._meta,
-    )
+    imput_meta = input_dask_ar._meta
+    if hasattr(imput_meta, "__array_function__"):
+        ar = np.zeros(
+            shape=(1, input_dask_ar.shape[1]),
+            dtype=input_dask_ar.dtype,
+            like=imput_meta,
+        )
+    elif "scipy.sparse" in type(imput_meta).__module__:
+        # sparse matrices dont support
+        # `like` due to non implimented __array_function__
+        # Refer https:/q/github.com/scipy/scipy/issues/10362
+        # Note below works for both cupy and scipy sparse matrices
+        ar = type(imput_meta)((1, input_dask_ar.shape[1]), dtype=input_dask_ar.dtype)
+    else:
+        msg = (
+            "\nYou did not provide metadata, so Dask is running the"
+            "function on a small dataset to guess output types. "
+            "It is possible that Dask will guess incorrectly.\n"
+            "To provide an explicit output types or to silence this message, "
```

Comment:
```suggestion
            "To provide an explicit output metadata to silence this message, "
```

I don't think "or" belongs in this sentence? Providing the explicit output metadata is the thing that silences this message, it's not two separate things, isn't that right? Also in this sentence it seems like `output types` => `output metadata`. 

If they are in fact two separate things, you'll need instructions on how to do both (eg: kwargs for providing explicit output type, as well as the kwargs for providing the output metadata)

Reply:
Addressed this with commit https://github.com/dask/dask-ml/pull/889/commits/48645448147126c6a343d6d085dd571be0cd1665